apps/nextjs/src/app/api/chat/chatHandler.ts

```diff
@@ -167,59 +171,227 @@ async function generateChatStream(
   );
 }
 
+function hasLessonPlan(obj: unknown): obj is { lessonPlan: unknown } {
+  return obj !== null && typeof obj === "object" && "lessonPlan" in obj;
+}
+
+function isValidLessonPlan(lessonPlan: unknown): boolean {
+  return lessonPlan !== null && typeof lessonPlan === "object";
+}
+
+function hasMessages(obj: unknown): obj is { messages: unknown } {
+  return obj !== null && typeof obj === "object" && "messages" in obj;
+}
+
+function isValidMessages(messages: unknown): boolean {
+  return Array.isArray(messages);
+}
+
+function verifyChatOwnership(
+  chat: { userId: string },
+  requestUserId: string,
+  chatId: string,
+): void {
+  if (chat.userId !== requestUserId) {
+    log.error(
+      `User ${requestUserId} attempted to access chat ${chatId} which belongs to ${chat.userId}`,
+    );
+    throw new Error("Unauthorized access to chat");
+  }
+}
+
+function parseChatOutput(
+  output: unknown,
+  chatId: string,
+): { messages: Message[]; lessonPlan: LooseLessonPlan } {
+  let messages: Message[] = [];
+  let lessonPlan: LooseLessonPlan = {};
+
+  try {
+    const parsedOutput =
+      typeof output === "string" ? JSON.parse(output) : output;
+
+    if (hasMessages(parsedOutput) && isValidMessages(parsedOutput.messages)) {
+      messages = parsedOutput.messages as Message[];
+    }
+
+    if (
+      hasLessonPlan(parsedOutput) &&
+      isValidLessonPlan(parsedOutput.lessonPlan)
+    ) {
+      lessonPlan = parsedOutput.lessonPlan as LooseLessonPlan;
+    }
+  } catch (error) {
+    log.error(`Error parsing output for chat ${chatId}`, error);
```

Comment:
I just want to confirm that we don't want to report to sentry here

Reply:
Thanks - I've added Sentry reporting. For now, no additional error handling for the front end though